src/irmin/content_addressable_intf.ml

```diff
@@ -27,14 +27,19 @@ module type S = sig
   include Read_only.S
   (** @inline *)
 
+  type hash
+  (** The type for keys hashes. *)
+
+  val index : [> read ] t -> hash -> key option Lwt.t
```

Comment:
missing docstring

Reply:
and it should say something about using the new optional `?index` argument that I just commented below :-)